From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: oop23 <ilidans01yt@gmail.com>
Date: Sat, 10 Oct 2026 04:36:17 +0000
Subject: [PATCH] Show positions as 1:25 / 1:05:30 instead of 1'25'' / 1h05'

Resume buttons and other positions used minute/second marks. They now
use the clock format every player uses. Also fixes the seconds-only
branch which compared milliseconds to 60.
---
 .../archos/mediacenter/utils/MediaUtils.java  | 34 ++++++-------------
 .../mediacenter/utils/FormatTimeTest.java     | 34 +++++++++++++++++++
 2 files changed, 45 insertions(+), 23 deletions(-)
 create mode 100644 test/java/com/archos/mediacenter/utils/FormatTimeTest.java

diff --git a/src/com/archos/mediacenter/utils/MediaUtils.java b/src/com/archos/mediacenter/utils/MediaUtils.java
index 231747f..580a15b 100644
--- a/src/com/archos/mediacenter/utils/MediaUtils.java
+++ b/src/com/archos/mediacenter/utils/MediaUtils.java
@@ -386,30 +386,18 @@ public static int restoreBestPosition(GridView view, int selectedPosition,
         }
     }
 
+    /**
+     * Clock style position, as every player shows it: "0:45", "1:25", "1:05:30".
+     * (Used to be "1'25''" / "1h05'".)
+     */
     static public String formatTime(long ms) {
-        String res;
-        if (ms <= 0)
-            res = "";
-        else {
-            // h = ms / 3600000;
-            // m = (ms % 3600000) / 60000;
-            // s = (ms % 60000) / 1000;
-            long sec = ms / 1000;
-            if (sec >= 3600) {
-                long m = (sec % 3600) / 60;
-                // %Hh%m'
-                res = String.valueOf(sec / 3600) + "h" + (m < 10 ? "0" + String.valueOf(m) : String.valueOf(m)) + "'";
-            }
-            else if (ms < 60)
-                // %s''
-                res = String.valueOf(sec % 60) + "''";
-            else {
-                long s = sec % 60;
-                // %m'%s''
-                res = String.valueOf((sec % 3600) / 60) + "'" + (s < 10 ? "0" + String.valueOf(s) : String.valueOf(s)) + "''";
-            }
-        }
-        return res;
+        if (ms <= 0) return "";
+        long sec = ms / 1000;
+        long h = sec / 3600;
+        long m = (sec % 3600) / 60;
+        long s = sec % 60;
+        if (h > 0) return String.format(java.util.Locale.ROOT, "%d:%02d:%02d", h, m, s);
+        return String.format(java.util.Locale.ROOT, "%d:%02d", m, s);
     }
 
     private static String mLastSdStatus;
diff --git a/test/java/com/archos/mediacenter/utils/FormatTimeTest.java b/test/java/com/archos/mediacenter/utils/FormatTimeTest.java
new file mode 100644
index 0000000..161add7
--- /dev/null
+++ b/test/java/com/archos/mediacenter/utils/FormatTimeTest.java
@@ -0,0 +1,34 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.archos.mediacenter.utils;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = Config.NONE, sdk = 33)
+public class FormatTimeTest {
+    @Test
+    public void clockStyle() {
+        assertEquals("", MediaUtils.formatTime(0));
+        assertEquals("0:45", MediaUtils.formatTime(45_000));
+        assertEquals("1:25", MediaUtils.formatTime(85_000));
+        assertEquals("59:59", MediaUtils.formatTime(3_599_000));
+        assertEquals("1:05:30", MediaUtils.formatTime(3_930_000));
+        assertEquals("2:00:00", MediaUtils.formatTime(7_200_400));
+    }
+}
